client/signup/steps/page-picker/index.tsx

```diff
@@ -199,36 +215,64 @@ const StyledButton = styled( Button )`
 	}
 `;
 
-export default function DIFMPagePicker( props: StepProps ) {
-	const { stepName, goToNextStep } = props;
-
+function DIFMPagePicker( props: StepProps ) {
+	const {
+		stepName,
+		goToNextStep,
+		signupDependencies: { siteId, siteSlug, newOrExistingSiteChoice },
+	} = props;
 	const translate = useTranslate();
 	const dispatch = useDispatch();
+	const [ isCheckoutPressed, setIsCheckoutPressed ] = useState( false );
 	const [ selectedPages, setSelectedPages ] = useState< string[] >( [
 		HOME_PAGE,
 		ABOUT_PAGE,
 		CONTACT_PAGE,
 	] );
+	const cartKey = useSelector( ( state ) => getSiteId( state, siteSlug ?? siteId ) );
```

Comment:
The cart key returns the site ID only when a site is selected. In the DIFM flow when we select the existing site we do not "select" the site, we just add it as a signupDependency. So we cannot use it here.

Reply:
I was referring to the `getCartKey` function here: https://github.com/Automattic/wp-calypso/blob/8a0f7c62407f7f13a92418f220a8e397a5704a55/client/my-sites/checkout/get-cart-key.ts#L4

This is a nit, so feel free to keep it as it is. I was thinking in terms of a future change, for example, if we change the cartKey to be something like `random-string-${siteId}`, it will be helpful to have this in one place.

Reply:
Ah sorry I was thinking about https://github.com/Automattic/wp-calypso/blob/e9d2f791cb8f207a10f80e251cec8e309e5d1963/client/my-sites/checkout/use-cart-key.ts#L7
This is a good suggestion I will include it!